src/apps/datasets/models.py

```diff
@@ -85,39 +84,29 @@ def in_use(self):
     def __str__(self):
         return self.name or ''
 
-    def get_chahub_endpoint(self):
+    @staticmethod
+    def get_chahub_endpoint():
         return "datasets/"
 
-    def clean_chahub_data(self, data):
-        validated_data = {}
-        for key, item in data.items():
-            if not item:
-                continue
-            elif isinstance(item, datetime.datetime):
-                validated_data[key] = item.isoformat()
-            elif isinstance(item, uuid.UUID):
-                validated_data[key] = str(item)
-            else:
-                validated_data[key] = item
-        return validated_data
+    def get_chahub_is_valid(self):
+        return self.upload_completed_successfully
+
+    def get_whitelist(self):
+        return ['remote_id', 'is_public']
 
     def get_chahub_data(self):
         data = {
             'creator_id': self.created_by.id,
             'remote_id': self.pk,
             'created_by': str(self.created_by.username),
-            'created_when': self.created_when,
+            'created_when': self.created_when.isoformat(),
             'name': self.name,
             'type': self.type,
             'description': self.description,
-            'key': self.key,
+            'key': str(self.key),
```

Comment:
is this missing `download_url`?

Reply:
Yes it was, hadn't been implemented yet. Is now.